server/sonar-server/src/main/java/org/sonar/server/license/ws/ListAction.java

```diff
@@ -0,0 +1,188 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.license.ws;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.sonar.api.PropertyType;
+import org.sonar.api.config.License;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.config.PropertyDefinitions;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.property.PropertyDto;
+import org.sonar.server.user.UserSession;
+import org.sonar.server.ws.WsAction;
+import org.sonarqube.ws.Licenses;
+import org.sonarqube.ws.Licenses.ListWsResponse;
+
+import static org.sonar.api.CoreProperties.PERMANENT_SERVER_ID;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.client.license.LicensesWsParameters.ACTION_LIST;
+
+public class ListAction implements WsAction {
+
+  private static final String ALL_SERVERS_VALUE = "*";
+
+  private final UserSession userSession;
+  private final PropertyDefinitions definitions;
+  private final DbClient dbClient;
+
+  public ListAction(UserSession userSession, PropertyDefinitions definitions, DbClient dbClient) {
+    this.userSession = userSession;
+    this.definitions = definitions;
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    context.createAction(ACTION_LIST)
+      .setDescription("List licenses settings.<br>" +
+        "Requires Administer System' permission")
+      .setResponseExample(getClass().getResource("list-example.json"))
+      .setSince("6.1")
+      .setInternal(true)
+      .setHandler(this);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    userSession.checkPermission(SYSTEM_ADMIN);
+
+    DbSession dbSession = dbClient.openSession(true);
+    try {
+      writeProtobuf(doHandle(dbSession), request, response);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private ListWsResponse doHandle(DbSession dbSession) {
+    Set<String> settingsKeys = definitions.getAll().stream()
+      .filter(definition -> definition.type().equals(PropertyType.LICENSE))
+      .map(PropertyDefinition::key)
+      .collect(Collectors.toSet());
+    settingsKeys.add(PERMANENT_SERVER_ID);
+    List<PropertyDto> properties = dbClient.propertiesDao().selectGlobalPropertiesByKeys(dbSession, settingsKeys);
+    return new ListResponseBuilder(settingsKeys, properties).build();
```

Comment:
In fact it's not possible without making ListResponseBuilder extends ListResponse, and I don't really like it...